include/deal.II/matrix_free/operators.h

```diff
@@ -717,6 +759,84 @@ namespace MatrixFreeOperators
     dst*= omega;
   }
 
+
+
+  //-----------------------------MassOperator----------------------------------
+
+  template <int dim, int fe_degree, int n_components, typename number>
+  MassOperator<dim, fe_degree, n_components, number>::
+  MassOperator (const MatrixFree<dim,number> &data)
+    :
+    Base<dim, number>()
+  {
+    Base<dim, number>::initialize(data);
+    compute_diagonal();
```

Comment:
since this class would be used by users, do not call `compute_diagonal()` within a constructor. Users may not need diagonal. The may only need `vmult`.